### ISSUE TITLE
Incorrect error message for unknown named character references

### ISSUE THREAD

**Issue by youming-lin** [mathiasbynens/he#51]
A minor issue:

`he.decode('&abc;', {strict: true})` throws error with this message: `Parse error: named character reference was not terminated by a semicolon`, when in fact neither `a` nor `ab` are valid legacy named character references and `&abc;` is terminated by `;`. I think an error message to the effect of `Parse error: named character reference is not spec-defined` would be better in this case.

This and #50 notwithstanding, `he` has been a great companion to the HTML5 spec as I learn about and write a spec-compliant HTML entity decoder for Swift :)

**Comment by mathiasbynens** [comment]
Another excellent report. Thank you, @youming-lin!

Will your decoder be open-source?

**Comment by youming-lin** [comment]
@mathiasbynens Yup, it is Apache 2.0 licensed: https://github.com/IBM-Swift/swift-html-entities

Version 1 supports HTML4 named character references; I'm currently working on version 2 which will support HTML5 named character references and improve parsing to detect parse errors as defined in the spec.

**Comment by messed-with** [comment]
I'm not a tech or anything. I've been hacked and you are followed by one of the team members. Can you help me? Im jenn. I dont get my emails. You can text or call 661.800.3964. I know...dont leave number. At this point idc. I just need help. Thanks.
Jenn
